modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/config/StatisticsObjectConfiguration.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.processors.query.stat.config;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.ignite.internal.processors.query.stat.StatisticsKey;
+import org.apache.ignite.internal.util.tostring.GridToStringInclude;
+import org.apache.ignite.internal.util.typedef.internal.S;
+
+/**
+ *
+ */
+public class StatisticsObjectConfiguration implements Serializable {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** */
+    @GridToStringInclude
+    private final StatisticsKey key;
+
+    /** */
+    @GridToStringInclude
+    private final StatisticsColumnConfiguration[] cols;
+
+    /** */
+    private final long ver;
+
+    /** */
+    public StatisticsObjectConfiguration(
+        StatisticsKey key,
+        StatisticsColumnConfiguration[] cols
+    ) {
+        this(key, cols, 0);
+    }
+
+    /** */
+    public StatisticsObjectConfiguration(
+        StatisticsKey key,
+        StatisticsColumnConfiguration[] cols,
+        long ver
+    ) {
+        this.key = key;
+        this.cols = cols;
+        this.ver = ver;
+    }
+
+    /** */
+    public static StatisticsObjectConfiguration merge(StatisticsObjectConfiguration oldCfg, StatisticsObjectConfiguration newCfg) {
+        assert oldCfg.key.equals(newCfg.key) : "Invalid schema to merge: [oldKey=" + oldCfg.key
+            + ", newKey=" + newCfg.key + ']';
+
+        Set<String> cols = Arrays.stream(oldCfg.cols)
+            .map(StatisticsColumnConfiguration::name).collect(Collectors.toSet());
+
+        cols.addAll(Arrays.stream(newCfg.cols)
+            .map(StatisticsColumnConfiguration::name).collect(Collectors.toSet()));
+
+        return new StatisticsObjectConfiguration(
+            newCfg.key,
+            cols.stream()
+                .map(StatisticsColumnConfiguration::new)
```

Comment:
Why not work with map<name, columnConfig> here? After any changes we'll have to rewrite it one more time to keep originals config instead just names. I suggest oldMap.putAll(newMap) will be enough here.